payment-method-messaging/src/main/java/com/stripe/android/paymentmethodmessaging/view/PaymentMethodMessagingView.kt

```diff
@@ -0,0 +1,356 @@
+        job = CoroutineScope(Dispatchers.IO).launch {
+            try {
+                val message = viewModel.loadMessage()
+                message.fold(
+                    onSuccess = {
+                        withContext(Dispatchers.Main) {
+                            data.value = PaymentMethodMessagingData(
+                                message = it,
+                                images = it.displayHtml.getBitmaps(this, stripeImageLoader),
+                                config = config
+                            )
+                            onSuccess()
+                        }
+                    },
+                    onFailure = {
+                        withContext(Dispatchers.Main) {
+                            onFailure(it)
+                        }
+                    }
+                )
+            } catch (e: CancellationException) {
+                onFailure(e)
+            }
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        job?.cancel()
+        super.onDetachedFromWindow()
+    }
+
+    data class Configuration @JvmOverloads constructor(
+        /**
+         * The publishable key used to make requests to Stripe.
+         */
+        internal val publishableKey: String,
+
+        /**
+         * The payment methods to display messaging for.
+         */
+        internal val paymentMethods: Set<PaymentMethod>,
+
+        /**
+         * The currency, as a three-letter [ISO currency code](https://www.iso.org/iso-4217-currency-codes.html).
+         */
+        internal val currency: String,
+
+        /**
+         * The purchase amount, in the smallest currency unit. e.g. 100 for $1 USD.
+         */
+        internal val amount: Int,
+
+        /**
+         * The current locale of the device.
+         */
+        internal val locale: Locale = Locale.current,
+
+        /**
+         * The customer's country as a two-letter string. Defaults to their device's country.
+         */
+        internal val countryCode: String = Locale.current.region,
+
+        /**
+         * The font of text displayed in the view. Defaults to the system font.
+         */
+        @FontRes internal val fontFamily: Int? = null,
+
+        /**
+         * The color of text displayed in the view.
+         */
+        @ColorInt internal val textColor: Int? = null,
+
+        /**
+         * The color of the images displayed in the view.
+         */
+        internal val imageColor: ImageColor? = null
+    ) {
+        /**
+         * Payment methods that can be displayed by `PaymentMethodMessagingView`
+         */
+        enum class PaymentMethod(internal val value: String) {
+            Klarna("klarna"),
+            AfterpayClearpay("afterpay_clearpay")
+        }
+
+        /**
+         * The colors of the image
+         */
+        enum class ImageColor(internal val value: String) {
+            Light("white"),
+            Dark("black"),
+            Color("color")
+        }
```

Comment:
During API review, we talked about making the naming a bit clearer (e.g. what is `ImageColor.Color`?). What happened with that?

Reply:
We haven't changed the API yet.